tests/basic/storage_pool_test.go

```diff
@@ -9894,3 +9898,219 @@ var _ = Describe("{AddDriveWithKernelPanic}", func() {
+					log.FailOnError(err, "error refreshing volume endpoints")
+					isjournal, err := IsJournalEnabled()
+					log.FailOnError(err, "Failed to check is journal enabled")
+					drvSize, err := getPoolDiskSize(poolListForOps[i])
+					log.FailOnError(err, "error getting drive size for pool [%s]", poolListForOps[i].Uuid)
+					drvMap, err := Inst().V.GetPoolDrives(&selectedNode)
+					log.FailOnError(err, "error in getting pool with IO with error")
+					if drvs, ok := drvMap[fmt.Sprintf("%d", poolListForOps[i].ID)]; ok {
+						if (POOL_MAX_CLOUD_DRIVES - len(drvs)) > 0 {
+							drivesThatCanBeAdded := (POOL_MAX_CLOUD_DRIVES - len(drvs))
+							for j := 1; j <= drivesThatCanBeAdded; j++ {
+								driveSize := drvSize * uint64(j)
+								expectedSize := (poolListForOps[i].TotalSize / units.GiB) + driveSize
+								expectedSizeWithJournal := expectedSize
+								if isjournal {
+									journalSize := uint64(3)
+									expectedSizeWithJournal = expectedSizeWithJournal - journalSize
+								}
+								err = Inst().V.ExpandPool(poolListForOps[i].Uuid, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, false)
+								log.FailOnError(err, "error while expanding pool")
+								maxDriveAllowed = maxDriveAllowed - 1
+								resizeErr := waitForPoolToBeResized(expectedSize, poolListForOps[i].Uuid, isjournal)
+								dash.VerifyFatal(resizeErr, nil, fmt.Sprintf("Expected new size to be '%d' or '%d'", expectedSize, expectedSize-3))
+								if maxDriveAllowed == 0 {
+									break
+								}
+							}
+						}
+					}
+				}
+			}
+			err = Inst().V.RefreshDriverEndpoints()
+			log.FailOnError(err, "error refreshing volume endpoints")
+			//get a pool where we can add one more drive
+			sNode, err := node.GetNodeByName(selectedNode.Name)
+			log.FailOnError(err, "failed while getting updated node pool list")
+			poolListOnNode, err := GetPoolsDetailsOnNode(sNode)
+			for i := 0; i < len(poolListOnNode); i++ {
+				drvMap, err := Inst().V.GetPoolDrives(&sNode)
+				log.FailOnError(err, "error getting pool drives from node [%s]", sNode.Name)
+				if drvs, ok := drvMap[fmt.Sprintf("%d", poolListOnNode[i].ID)]; ok {
+					if (POOL_MAX_CLOUD_DRIVES - len(drvs)) > 0 {
+						selectedPools = poolListOnNode[i].Uuid
+						break
+					}
+				}
+			}
+			//check if selectedpool is empty
+			if selectedPools == "" {
+				log.FailOnError(fmt.Errorf("failed to get a pool to add a drive"), "failed to get a pool to add a drive")
+			}
+			//add one more extra drive than allowed per node expect error
+			poolToBeResized, err := GetStoragePoolByUUID(selectedPools)
+			log.FailOnError(err, "failed to get a pool to add a drive")
+			drvSize, err := getPoolDiskSize(poolToBeResized)
+			log.FailOnError(err, "error getting drive size for pool [%s]", selectedPools)
+			expectedSize := (poolToBeResized.TotalSize / units.GiB) + drvSize
+			err = Inst().V.ExpandPool(poolToBeResized.Uuid, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, false)
```

Comment:
please handle case when volume is not clean ? based on err if volume is not clean please expand pool with -f options